chatsky/context_storages/database.py

```diff
@@ -8,181 +8,135 @@
 
-    def __contains__(self, key: Hashable) -> bool:
-        """
-        Synchronous method for finding whether any Context is stored with given key.
-
-        :param key: Hashable key used to check if Context instance is stored.
-        :return: True if there is Context accessible by given key, False otherwise.
-        """
-        return asyncio.run(self.contains_async(key))
-
     @abstractmethod
-    async def contains_async(self, key: Hashable) -> bool:
+    async def load_field_latest(self, ctx_id: str, field_name: str) -> List[Tuple[Hashable, bytes]]:
         """
-        Asynchronous method for finding whether any Context is stored with given key.
-
-        :param key: Hashable key used to check if Context instance is stored.
-        :return: True if there is Context accessible by given key, False otherwise.
+        Load the latest field data.
         """
         raise NotImplementedError
 
-    def __len__(self) -> int:
-        """
-        Synchronous method for retrieving number of stored Contexts.
-
-        :return: The number of stored Contexts.
-        """
-        return asyncio.run(self.len_async())
-
     @abstractmethod
-    async def len_async(self) -> int:
+    async def load_field_keys(self, ctx_id: str, field_name: str) -> List[Hashable]:
         """
-        Asynchronous method for retrieving number of stored Contexts.
-
-        :return: The number of stored Contexts.
+        Load all field keys.
         """
         raise NotImplementedError
 
-    def get(self, key: Hashable, default: Optional[Context] = None) -> Context:
-        """
-        Synchronous method for accessing stored Context, returning default if no Context is stored with the given key.
-
-        :param key: Hashable key used to store Context instance.
-        :param default: Optional default value to be returned if no Context is found.
-        :return: The stored context, associated with the given key or default value.
-        """
-        return asyncio.run(self.get_async(key, default))
-
-    async def get_async(self, key: Hashable, default: Optional[Context] = None) -> Context:
+    @abstractmethod
+    async def load_field_items(self, ctx_id: str, field_name: str, keys: Set[Hashable]) -> List[bytes]:
         """
-        Asynchronous method for accessing stored Context, returning default if no Context is stored with the given key.
-
-        :param key: Hashable key used to store Context instance.
-        :param default: Optional default value to be returned if no Context is found.
-        :return: The stored context, associated with the given key or default value.
+        Load field items.
         """
-        try:
-            return await self.get_item_async(str(key))
-        except KeyError:
-            return default
+        raise NotImplementedError
 
-    def clear(self):
+    @abstractmethod
+    async def update_field_items(self, ctx_id: str, field_name: str, items: List[Tuple[Hashable, bytes]]) -> None:
```

Comment:
Why is `items` not a dict?

Reply:
Do you mean for compatibility with other `DBContextStorage` methods?
My suggestion was to make every method that returns and/or accepts `List[Tuple[Hashable, Any]]` to instead return and/or accept `Dict[Hashable, Any]`.